### PR TITLE
Implements tracing capabilities through telemetry compiler feature flag

### PR DESCRIPTION
**Pull request by FlorentinDUBOIS**
Hello :wave:, 

To find an issue on production, I needed more observability of the software that use this crate, to get more insights, I put tracing annotation on functions behind a compiler feature flag to not add change to the current behavior of the driver.

Besides, I saw that the driver needed some love. So, I have updated all dependencies using the same way to write the semantic version using `^x.x.x` which allow to lock easily a version, if something does not work as expected in future version of a crate through this syntax `=x.x.x`. I also have updated the Rust edition which is now the latest one `2021`, which asked at least Rust `1.56.0`, if my memory is good.

Last thing, I have set the right version number of the crate on the `Cargo.toml` according to the latest git tag.

Have a wonderful day!

### CODE REVIEW

**Comment by FlorentinDUBOIS** [comment]
Besides, if this pull reques is merged, it closes #222

File: Cargo.toml
Comment (KannarFr) [inline Cargo.toml:4]: these should be done in the release commit after merging.

**Review by KannarFr — CHANGES_REQUESTED** [review]
Just a strange version change in a PR :p, otherwise LGTM.

File: Cargo.toml
Comment (CleverAkanoa) [reply Cargo.toml:4]: I agree

Comment (FlorentinDUBOIS) [reply Cargo.toml:4]: I agree, but the version 4.1.3 is already published according to git tag. I will create another commit to do this change.

Comment (FlorentinDUBOIS) [reply Cargo.toml:4]: I have remove the commit that do this change

**Comment by FlorentinDUBOIS** [comment]
@KannarFr @CleverAkanoa could you approve if it is ok for you?
@utahkay @vmalloc do you agree with those changes?

**Comment by vmalloc** [comment]
Not sure I'm the right person to ask...

**Comment by utahkay** [comment]
I'm not sure I'm the right person to ask, either...

**Review by CleverAkanoa — APPROVED** [review]
Seems good for me.

Maybe we can document this telemetry feature in the Readme file.

**Comment by FlorentinDUBOIS** [comment]
You are right, I will do that 😇

**Pull request merged**
